rotkehlchen/chain/evm/decoding/zero_x/decoder.py

```diff
@@ -0,0 +1,208 @@
+                not self.base.is_tracked(string_to_evm_address(event.location_label))
+            ):
+                continue
+
+            if ((
+                event.event_type == HistoryEventType.TRADE and  # for token to token
+                event.event_subtype == HistoryEventSubType.SPEND
+            ) or (
+                event.event_type == HistoryEventType.SPEND and  # for eth to token
+                event.event_subtype == HistoryEventSubType.NONE
+            )) and event.address is not None:
+                sent_asset = event.asset
+                send_address_to_events[event.address] = event
+
+            if ((
+                event.event_type == HistoryEventType.TRADE and  # for token to token
+                event.event_subtype == HistoryEventSubType.RECEIVE
+            ) or (
+                event.event_type == HistoryEventType.RECEIVE and  # for token to eth
+                event.event_subtype == HistoryEventSubType.NONE
+            )) and event.address is not None:
+                received_asset = event.asset
+                receive_address_to_events[event.address] = event
+
+        if sent_asset is None or received_asset is None:
+            return decoded_events
+
+        for zero_x_address in (self.router_address, self.flash_wallet_address):
+            if (
+                not sent_asset.is_evm_token() and  # sent_asset is native currency
+                zero_x_address in send_address_to_events and  # is sent to 0x
+                send_address_to_events[zero_x_address].event_type == HistoryEventType.SPEND and
+                send_address_to_events[zero_x_address].event_subtype == HistoryEventSubType.NONE
+            ):
+                self._update_send_and_receive_events(
+                    send_event=send_address_to_events[zero_x_address],
+                )
+
+            if (
+                not received_asset.is_evm_token() and  # received_asset is native_currency
+                zero_x_address in receive_address_to_events and  # is received from 0x
+                receive_address_to_events[zero_x_address].event_type == HistoryEventType.RECEIVE and  # noqa: E501
+                receive_address_to_events[zero_x_address].event_subtype == HistoryEventSubType.NONE
+            ):
+                self._update_send_and_receive_events(
+                    receive_event=receive_address_to_events[zero_x_address],
+                )
+
+        for _log in all_logs:
+            send_event = send_address_to_events.get(_log.address)
+            receive_event = receive_address_to_events.get(_log.address)
+
+            if _log.topics[0] in {
+                UNISWAP_V2_SWAP_SIGNATURE,  # contract is Uniswap V2
+                UNISWAP_V3_SWAP_SIGNATURE,  # contract is Uniswap V3
+            } and self.router_address in {
+                hex_or_bytes_to_address(_log.topics[1]),  # 0x is sender
+                hex_or_bytes_to_address(_log.topics[2]),  # 0x is receiver
+            }:
```

Comment:
Make this a constant final somewhere

```suggestion
            if _log.topics[0] in UNISWAP_SIGNATURES and 
            } and self.router_address in {
                hex_or_bytes_to_address(_log.topics[1]),  # 0x is sender
                hex_or_bytes_to_address(_log.topics[2]),  # 0x is receiver
            }:
```